lib/semian.rb

```diff
@@ -156,6 +157,14 @@ def register(name, **options)
   end
 
   def retrieve_or_register(name, **args)
+    # If consumer who retrieved / registered by a Semian::Adapter, keep track
+    # of who the consumer was so that we can clear the resource reference if needed.
+    if consumer = args.delete(:consumer)
+      if consumer.class.included_modules.include?(Semian::Adapter)
```

Comment:
None, except it's slower because it has to allocate an Array.

The proper way to do this is `consumer.is_a?(Semian::Adapter)`